Modified patch from RHEL3 63.EL - check that the socket is still connected

 net/sunrpc/svcsock.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/net/sunrpc/svcsock.c	2010-02-02 00:04:46.000000000 +0300
+++ b/net/sunrpc/svcsock.c	2010-07-23 11:57:09.000000000 +0400
@@ -350,6 +350,10 @@ svc_sendto(struct svc_rqst *rqstp, struc
 	 */
 	msg.msg_flags	= 0;
 
+	/* Make sure there is a socket before sending */
+	if (test_bit(SK_DEAD, &svsk->sk_flags))
+		return -ENOTCONN;
+
 	oldfs = get_fs(); set_fs(KERNEL_DS);
 	len = sock_sendmsg(sock, &msg, buflen);
 	set_fs(oldfs);
